android/src/main/kotlin/africa/ejara/trustdart/coins/XTZ.kt

```diff
@@ -31,4 +40,159 @@ class XTZ : Coin("XTZ", CoinType.TEZOS) {
+        passphrase: String
+    ): String? {
+        val cmd: String? = txData["cmd"] as String?
+        val wallet = HDWallet(mnemonic, passphrase)
+        val privateKey = wallet.getKey(coinType, path)
+        val txHash: String?
+        val publicKey = privateKey.publicKeyEd25519.data()
+        when (cmd) {
+            "FA2" -> {
+                val transferInfos = Txs.newBuilder()
+                    .setAmount(txData["amount"] as String)
+                    .setTokenId(txData["tokenId"] as String)
+                    .setTo(txData["toAddress"] as String)
+                    .build()
+
+                val txObj = TxObject.newBuilder()
+                    .setFrom(txData["senderAddress"] as String)
+                    .addTxs(transferInfos)
+                    .build()
+
+                val fa2 = FA2Parameters.newBuilder()
+                    .setEntrypoint("transfer")
+                    .addTxsObject(txObj)
+                    .build()
+
+                val parameters = OperationParameters.newBuilder()
+                    .setFa2Parameters(fa2)
+                    .build()
+
+                val transactionData = TransactionOperationData.newBuilder()
+                    .setAmount(txData["transactionAmount"]!!.toLong())
+                    .setDestination(txData["destination"] as String)
+                    .setParameters(parameters)
+                    .build()
+
+                val transaction = Operation.newBuilder()
+                    .setSource(txData["source"] as String)
+                    .setFee(txData["fee"]!!.toLong())
+                    .setCounter(txData["counter"]!!.toLong())
+                    .setGasLimit(txData["gasLimit"]!!.toLong())
+                    .setStorageLimit(txData["storageLimit"]!!.toLong())
+                    .setKind(Operation.OperationKind.TRANSACTION)
+                    .setTransactionOperationData(transactionData)
+                    .build()
+
+                val operationList = OperationList.newBuilder()
+                    .setBranch(txData["branch"] as String)
+                    .addOperations(transaction)
+                    .build()
+
+                if(txData["revealed"] == false){
+                    val revealOperationData = RevealOperationData.newBuilder()
+                        .setPublicKey(ByteString.copyFrom(publicKey))
+
+                    val reveal = Operation.newBuilder()
+                        .setSource(txData["source"] as String)
+                        .setFee(txData["fee"]!!.toLong())
+                        .setCounter(txData["counter"]!!.toLong())
+                        .setGasLimit(txData["gasLimit"]!!.toLong())
+                        .setStorageLimit(txData["storageLimit"]!!.toLong())
+                        .setKind(Operation.OperationKind.REVEAL)
+                        .setRevealOperationData(revealOperationData)
+                        .build()
+
+                    operationList.operationsList.add(reveal)
```

Comment:
fee, counter, gasLimit and storageLimit are different for reveal so have their versions.